[FEAT] iOS 3주차 과제

noah0316 opened the pull request:
## 📌 관련 이슈
<!-- 관련있는 이슈 번호(#000)을 적어주세요.
  해당 pull request merge와 함께 이슈를 닫으려면 
  closed #Issue_number를 적어주세요 -->
iOS 3주차 과제를 진행했습니다!!

- linked: #11
## 📌 변경 사항 및 이유
<!-- 변경한 내용과 그 이유를 적어주세요. -->
3주차 과제는 TableView와 CollectionView의 적용입니다!

## 📌 PR Point
<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
그동안 코드를 너무 중구난방으로 짰다는 걸 느껴, 
이제라도  MVC 패턴을 적용하며 연습해보고 싶어 차근차근 적용중입니다!
도전과제도 언넝 해보도록 하겠습니다!

noah0316 committed:
💄[MOD] : TableView 생성 및 연결

noah0316 committed:
🍱[ADD] : 3주차 과제 에셋 추가

noah0316 committed:
✨[FEAT] : TableViewCell 구현

noah0316 committed:
➕[ADD] : TableView 관련모델 추가
- HomeManager.swift: Thumbnail 모델과, Channel 모델을 관리해주는
  HomeManager 추가
- Thumbnail.swift: TableView 관련 모델 추가

noah0316 committed:
➕[ADD] : UITableView Extension 추가 - HomeVC
- HomeVC+TableView.swift : UITableViewDelegate, UITableViewDataSource 구현

noah0316 committed:
✨[FEAT] : CollectionViewCell 구현

noah0316 committed:
➕[ADD] : CollectionView 관련모델 추가
- Channel.swift: CollectionView 관련모델 Channel추가

noah0316 committed:
➕[ADD] : UICollectionView Extension 추가 - HomeVC
- HomeVC+CollectionView.swift: UICollectionViewDelegate,
  UICollectionViewDataSource, UICollectionViewDelegateFlowLayout 구현

noah0316 committed:
✨[FEAT] : 3주차과제 UICollectionView, UITableView 구현

yangsubinn reviewed (approved):
전 승현님 코드 볼때마다 놀라여.... 항상 많이 배웁니다.... 감사함다 수고하셨습니다아🙇‍♀️

hellozo0 commented on Assignment/29th-assignment/29th-assignment/Screens/Home/Controller/HomeVC+CollectionView.swift:
```diff
@@ -0,0 +1,43 @@
+
+import UIKit
+
+extension HomeVC: UICollectionViewDelegate {
```
extension 파일로 따로 빼서 관리하고 싶었는는데 실천한 승현슨배.. 멋잇다...

hwangJi-dev replied:
와 extension만 한 파일로 관리라니 멋지다....

noah0316 replied:
슨배님들이 더더더 멋있고..🥇🥇🥇

hellozo0 commented on Assignment/29th-assignment/29th-assignment/Screens/Home/Model/Thumbnail.swift:
```diff
@@ -0,0 +1,18 @@
+
+struct Thumbnail  {
+	var thumbNailImage: UIImage
+	var	profileImage: UIImage
```
ㅋㅋㅋㅋㅋ 빈칸지적 넘 웃겨요 수빈슨배

hwangJi-dev replied:
빈칸ㅋㅋㅋㅋㅋㅋㅋㅋㅋ

hellozo0 reviewed (approved):
승현슨배 넘 고생했다!
도전과제 언능 후루룩 뚝딱 할 수 있을거야! extension파일 따로 뺀거 보고 아..나도 빨리 해봐야겠다... 
라는 생각을 했다,, 역시,,, 당신,,, YB 생태계 파괴자... 멈춰..!

hwangJi-dev commented on Assignment/29th-assignment/29th-assignment/Screens/Home/Controller/HomeVC.swift:
```diff
@@ -9,21 +9,36 @@ import UIKit
-    override func viewDidLoad() {
+	var manager = HomeManager.shared
+	
+	@IBOutlet weak var tableView: UITableView! {
+		didSet {
+			tableView.delegate = self
+			tableView.dataSource = self
+			registerTableViewXib()
+		}
+	}
+	
```
didSet 활용 넘 조아요~~~~!

noah0316 replied:
이 `didSet` 활용은 지은 선배로 부터 시작되어 일년에 한바퀴 돌면서 받는 사람에게 행운을 주었고..

hwangJi-dev commented on Assignment/29th-assignment/29th-assignment/Screens/Home/Model/HomeManager.swift:
```diff
@@ -0,0 +1,84 @@
+
+import UIKit
+
+struct HomeManager {
```
아니 이 파일 뭐냐고,, 진짜 유용성 몬데... 그리고 다음에 Swift 문법 강의 해주면 안돼 ...? ? ? ? ? ? ? ?

yangsubinn replied:
@noah0316 진짜로... 문법 강의 좀 해주세요.... 감사합니다~🍎

noah0316 replied:
윤아슨배 싱글톤이 너무 멋져보여서 쇽샥 해보았을 뿐이고..🤣🤣

hwangJi-dev reviewed (approved):
갓승현은 당장 금잔디파를 대상으로 swift 문법 강의를 해주어라!
진짜 대박쓰,, 오늘도 많이 배워감~~~~~!!!!

noah0316 committed:
⚰️ [DEL] : 불필요한 import문 제거

noah0316 committed:
➕[ADD] : UIView extension 추가
- UIView+ : UI관련 설정을 IB에서 처리하기 위한 UIView 기능확장(border, color)

noah0316 committed:
➕[ADD] : Cell이 채택해야 하는 프로토콜 선언
- CellProtocol : 제네릭을 사용하는 곳에서 custom type, method를 더
  이용하기 쉽게 추가해보았습니다.

noah0316 committed:
➕[ADD] : Manager 프로토콜 선언
- ManagerProtocol : extension을 이용해 기능을 확장할 때 어떤 Manager와
  관련된 extension인지 더 명확하게 표기하기 위해 프로토콜을
  선언해주었습니다.

noah0316 committed:
✅[CHORE] : Manager, CellProtocol 프로토콜 채택
- ChannelCVC, ThumbnailTVC, HomeManager

noah0316 committed:
➕[ADD] : single ton 객체 private init() 추가
- Swift에서 전역변수를 사용하여 인스턴스를 생성하면, 사용시점에 초기화
  되고, GCD의 dispatch_once가 자동적용되어 Thread-Safe하게 사용할 수
  있다고 합니다!(Swift 3.0+)

noah0316 committed:
💄[MOD] : 도전과제 UI구현

noah0316 committed:
✨[FEAT] : 도전과제 기능 구현

Suyeon9911 commented:
슬쩍 구경하고 갑니다.. 와비 생태계 파괴자 인정... ㅎ !!!

noah0316 commented:
> 슬쩍 구경하고 갑니다.. 와비 생태계 파괴자 인정... ㅎ !!!

@Suyeon9911 어훟ㅎㅎ아직 한참 멀었습니다..구경와주셔서 너무 감사함다 ㅎㅎ🙇‍♂️🙇‍♂️
